src/cloudai/workloads/slurm_ray_container/slurm_command_gen_strategy.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+# Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Any, Dict, List, Union, cast
+
+from jinja2 import Template
+
+from cloudai import TestRun
+from cloudai.workloads.slurm_container import SlurmContainerCommandGenStrategy
+
+from .slurm_ray_container import SlurmRayContainerTestDefinition
+
+
+class SlurmRayContainerCommandGenStrategy(SlurmContainerCommandGenStrategy):
+    """Command generation strategy for generic Slurm container tests."""
+
+    def _get_sbatch_directives(self, args: Dict[str, Any], output_path: Path) -> Dict[str, str]:
+        sbatch_directives = super()._get_sbatch_directives(args, output_path)
+        # TODO(Amey): We probably need to figure out what to do with cpus-per-task, mem-per-cpu
```

Comment:
This can be set with `SlurmSystem.extra_sbatch_args`. The downside is that it is set per System, so all tests in a scenario will have it.

Reply:
Basically, i want this to be dynamic, as a fraction of total resources

Reply:
Since we have to set the task per worker to 1 for ray, we need to ensure that all the resources are made available to the process.